qkit/drivers/HP_3245A.py

```diff
@@ -0,0 +1,188 @@
+# HP_3245A.py driver for Hewlett Packard 3245A Universal Source (current / voltage).
+# Sergey Danilin @University of Glasgow, 01/2020
+
+from qkit.core.instrument_base import Instrument
+from qkit import visa
+import types
+import logging
+import numpy as np
+
+class HP_3245A(Instrument):
+    '''
+	This is the python driver for the Hewlett Packard 3245A Universal Source
+
+	Usage: perform current or voltage sweep
+	Initialise with
+	<name> = qkit.instruments.create('<name>', 'HP_3245A', address='<GPIB address>', reset=<bool>)
+	
+	'''
+    def __init__(self, name, address, reset = False):
+        '''
+        Initializes the HP 3245A source, and communicates with the wrapper.
+
+        Input:
+            name (string)    : name of the instrument
+            address (string) : GPIB address
+            reset (bool)     : resets to default values
+
+        Output:
+            None
+        '''
+        # Initialize wrapper functions
+        logging.info(__name__ + ' : Initializing instrument HP_3245A')
+        Instrument.__init__(self, name, tags=['physical'])
+                
+        self._address = address
+        self._visainstrument = visa.instrument(self._address)
+      
+        self.add_function('reset')
+        self.add_function('clear_memory')
+        self.add_function('set_channel')
+        self.add_function('set_terminal')
+        self.add_function('set_output_type')
+        self.add_function('set_impedance')
+        self.add_function('set_resolution')
+        self.add_function('set_autorange')
+        self.add_function('set_current')
+        self.add_function('set_voltage')
+
```

Comment:
Would it make sense to add current, voltage and maybe resolution as instrument parameters? This has the advantage that the values are stored in each measurement .h5 file.

Reply:
Thanks for the comments! And sorry I only saw them yesterday.
Yes, I think here we could definitely make them instrument parameters.